### ISSUE TITLE
`Deposits may be interrupted while your WooPayments balance remains negative` message erroneously shown for instant payouts merchants

### ISSUE THREAD

**Issue by csmcneill** [Automattic/woocommerce-payments#6941]
### Describe the bug
When an instant deposit is initiated, there's a transfer for the instant deposit fee.

This can produce a negative `Available funds` balance at **Payments > Overview.** This will trigger the following notice on the same page in the `Deposits` card...

```
Deposits may be interrupted while your WooPayments balance remains negative.
```

...even though the negative `Available funds` balance is countered by a positive `Pending funds` balance.

### To Reproduce
I haven't reproduced this personally yet, but theoretically:

1. Purchase goods/services.
2. Trigger an instant deposit.
3. Check the `Deposits` card.
4. Notice the message about the negative balance.

### Actual behavior
Instant Deposits is a feature of WooPayments, and the transfer associated with it can erroneously display that an account has a negative balance temporarily. This can lead to confusion for merchants.

### Screenshots

![Markup on 2023-08-06 at 15:08:30](https://github.com/Automattic/woocommerce-payments/assets/82999806/9e162b68-4330-4d27-a889-5cb08da99810)

### Expected behavior

This information is pulled directly from the Stripe dashboard and should be temporary while funds are moved around. However, I think that the negative balance notice here should only show if `Available funds` + `Pending funds` < `0`. Currently, the message appears to only rely on the `Available funds` balance.

### Additional context

6614021-zen

**Comment by haszari** [comment]
TLDR: the `Deposits may be interrupted while your WooPayments balance remains negative.` notice is erroneously shown to merchants. 

We need to ensure this is only shown when merchant's payouts are really paused. 

> This information is pulled directly from the Stripe dashboard and should be temporary while funds are moved around. However, I think that the negative balance notice here should only show if Available funds + Pending funds < 0. Currently, the message appears to only rely on the Available funds balance.

☝️ great idea @csmcneill

**Comment by haszari** [comment]
Re-tagging this as maintenance. We will be working on instant deposits soon so now's a good time to fix this. 

Added `high` priority since this message erodes trust in the system, and in particular, could discourage use of instant deposits. 

This is related to #8070 – i.e. the way we present balances leads to edge-cases like this. Prioritising alongside that issue.
